compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1965,34 +1968,34 @@ pub struct TargetOptions {
     linker_is_gnu_json: bool,
 
     /// Objects to link before and after all other object code.
-    pub pre_link_objects: CrtObjects,
-    pub post_link_objects: CrtObjects,
+    pub pre_link_objects: MaybeLazy<CrtObjects>,
```

Comment:
`LinkArgs` and `CrtObjects` are always supposed to be lazy, so `MaybeLazy` can be moved into the type alias themselves.